examples/misc/MonadicLetBindings.fst

```diff
@@ -81,3 +138,15 @@ let rec eval (e: exp): option int
                  | y -> let? x = eval x in
                        Some (x / y)
 
+
+(**** [pair] is just a reformulation of [<*>] *)
+// Section 7 of McBride & Patterson "Applicative programming with
+// effects" [https://www.staff.city.ac.uk/~ross/papers/Applicative.pdf]
+let applicative_eq (m: Type -> Type) (fmap: (#a:Type -> #b:Type -> (a -> b) -> m a -> m b)) =
+  let (<*>) (pair: (#a:Type -> #b:Type -> m a -> m b -> m (a * b)))
+    : #a:Type -> #b:Type -> m (a -> b) -> m a -> m b
+    = fun f o -> fmap (fun (f, x) -> f x) (pair f o) in
+  let pair ((<*>): (#a:Type -> #b:Type -> m (a -> b) -> m a -> m b))
+    : #a:Type -> #b:Type -> m a -> m b -> m (a * b)
+    = fun x y -> fmap Mktuple2 x <*> y in
```

Comment:
Yeah, maybe it's a little off-topic.
I wrote that to explain the statement:
https://github.com/FStarLang/FStar/blob/0134b251b0889ddce0dc78840f2afadfa973a148/examples/misc/MonadicLetBindings.fst#L15-L17
I was thinking that the equivalence between such a `pair` and a `<*>` operator is not very intuitive, so showing it explicitly might help.

Should I remove it? Otherwise, I can add a "see [applicative_eq] at the end of the file" to the comment "a reformulation of the [<*>] operator of an applicative" (so that this last chunk of code feels less like coming from nowhere!)

Reply:
it's fine as is, don't worry about it.